harness/determined/deploy/aws/aws.py

```diff
@@ -317,11 +353,30 @@ def deploy_stack(
     if stack_exists(stack_name, boto3_session):
         print("True - Updating Stack")
 
-        update_stack(stack_name, template_body, boto3_session, parameters)
+        if not no_prompt:
+            tags = get_tags(stack_name, boto3_session)
+            prompt_needed = False
+            if constants.deployment_types.TYPE_TAG_KEY not in tags:
+                print("Previous value of --deployment-type is not known.")
+                prompt_needed = True
+            elif tags[constants.deployment_types.TYPE_TAG_KEY] != deployment_type:
+                print("Value of --deployment-type has changed!")
+                prompt_needed = True
+
+            if prompt_needed:
+                val = input(
+                    "If --deployment-type has changed, updating the stack may erase the database. "
+                    "Are you sure you want to proceed? [y/N]"
+                )
+                if val.lower() != "y":
+                    print("Update cancelled.")
```

Comment:
I think the american spelling is "canceled" with one `l`.  Not 100% sure though.

Reply:
22 1/2 years here, and this is actually a difference I hadn't noticed before...